[docs] Add section to devguide on build targets (#12513)
elastic/beats: docs/devguide/contributing.asciidoc

Virtualenv can be installed with the command `easy_install virtualenv` or `pip
install virtualenv`. More details can be found
https://virtualenv.pypa.io/en/latest/installation.html[here]. Both of these commands should be run before submitting a PR. You can view all the available make targets with `make help`.

[float]
[[build-target-env-vars]]
=== Selecting Build Targets

Beats is built using the `make release` target. By default, make will select from a limited number of default build targets:

- darwin/amd64
- linux/386 
- linux/amd64
- windows/386 
- windows/amd64

> You can change build targets using the `PLATFORMS` environment variable. Targets set with the `PLATFORMS` variable can either be a GOOS value, or a GOOS/arch pair. For example, `linux` and `linux/amd64` are both valid targets. You can select multiple targets, and the `PLATFORMS` list is space delimited, for example `darwin windows` will build on all supported darwin and windows architectures. In addition, you can add or remove from the list of build targets by prepending `+` or `-` ot a given target, as such: `+bsd` or `-darwin`.

Reviewer: ```suggestion
You can change build targets using the `PLATFORMS` environment variable. Targets set with the `PLATFORMS` variable can either be a GOOS value, or a GOOS/arch pair. For example, `linux` and `linux/amd64` are both valid targets. You can select multiple targets, and the `PLATFORMS` list is space delimited, for example `darwin windows` will build on all supported darwin and windows architectures. In addition, you can add or remove from the list of build targets by prepending `+` or `-`to a given target. For example: `+bsd` or `-darwin`.
```

Reviewer: I'd recommend breaking the line at ~85 char for better diffs in GitHub.

Reviewer: I'm assuming readers will understand what you mean by a GOOS value? I don't, but don't build platform-specific binaries either. :-P

Author: I think that's fairly well understood? And yah, that's a somewhat long line.

Virtualenv can be installed with the command `easy_install virtualenv` or `pip
install virtualenv`. More details can be found
https://virtualenv.pypa.io/en/latest/installation.html[here]. Both of these commands should be run before submitting a PR. You can view all the available make targets with `make help`.

[float]
[[build-target-env-vars]]
=== Selecting Build Targets

Beats is built using the `make release` target. By default, make will select from a limited number of preset build targets:

- darwin/amd64
- linux/386 
- linux/amd64
- windows/386 
- windows/amd64

You can change build targets using the `PLATFORMS` environment variable. Targets set with the `PLATFORMS` variable can either be a GOOS value, or a GOOS/arch pair. 
For example, `linux` and `linux/amd64` are both valid targets. You can select multiple targets, and the `PLATFORMS` list is space delimited, for example `darwin windows` will build on all supported darwin and windows architectures. 

> In addition, you can add or remove from the list of build targets by prepending `+` or `-` ot a given target, as such: `+bsd` or `-darwin`.

Reviewer: ```suggestion
In addition, you can add or remove from the list of build targets by prepending `+` or `-` to a given target. For example: `+bsd` or `-darwin`.
```

Reviewer: My suggestions got swallowed, so made them again. :-)

Author: Ah yah, hah, the huge line made it hard to see what was going on, hah
